artifactory/services/buildinfo.go

```diff
@@ -87,3 +95,37 @@ func (bis *BuildInfoService) PublishBuildInfo(build *buildinfo.BuildInfo, projec
 	log.Debug("Artifactory response:", resp.Status)
 	return summary, nil
 }
+
+func (bis *BuildInfoService) DeleteBuildInfo(build *buildinfo.BuildInfo, projectKey string, buildNumberFrequency int) error {
+	params := CreateDeleteBuildInfoBody(build, projectKey, buildNumberFrequency)
+	content, err := json.Marshal(params)
+	if err != nil {
+		return err
+	}
+	httpClientsDetails := bis.GetArtifactoryDetails().CreateHttpClientDetails()
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+	resp, body, err := bis.client.SendPost(bis.GetArtifactoryDetails().GetUrl()+"api/build/delete", content, &httpClientsDetails)
+	if err != nil {
+		log.Error("Error occurred while deleting build info", err)
+		return err
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK, http.StatusCreated, http.StatusNoContent); err != nil {
+		return err
+	}
+	log.Debug("Artifactory response", resp.Status)
+	return nil
+}
+
+func CreateDeleteBuildInfoBody(build *buildinfo.BuildInfo, projectKey string, buildNumberFrequency int) DeleteBuildInfoBody {
+	buildNumbers := make([]string, 0, buildNumberFrequency)
+	for i := 0; i < buildNumberFrequency; i++ {
```

Comment:
I can't seem to understand what `buildNumberFrequency` does from the code, and there is no documentation on it. Please add some comments and documentation in the README regarding this input. We can also first discuss the functionality it adds over the REST API if you prefer.